tests/TestHarness/Node.py

```diff
@@ -1017,20 +1020,29 @@ def killNodeOnProducer(self, producer, whereInSequence, blockType=BlockType.head
         assert(isinstance(blockType, BlockType))
         assert(isinstance(returnType, ReturnType))
         basedOnLib="true" if blockType==BlockType.lib else "false"
-        payload="{ \"producer\":\"%s\", \"where_in_sequence\":%d, \"based_on_lib\":\"%s\" }" % (producer, whereInSequence, basedOnLib)
-        return self.processCurlCmd("test_control", "kill_node_on_producer", payload, silentErrors=silentErrors, exitOnError=exitOnError, exitMsg=exitMsg, returnType=returnType)
-
-    def processCurlCmd(self, resource, command, payload, silentErrors=True, exitOnError=False, exitMsg=None, returnType=ReturnType.json):
-        cmd="curl %s/v1/%s/%s -d '%s' -X POST -H \"Content-Type: application/json\"" % \
-            (self.endpointHttp, resource, command, payload)
+        payload={ "producer":producer, "where_in_sequence":whereInSequence, "based_on_lib":basedOnLib }
+        return self.processUrllibRequest("test_control", "kill_node_on_producer", payload, silentErrors=silentErrors, exitOnError=exitOnError, exitMsg=exitMsg, returnType=returnType)
+
+    def processUrllibRequest(self, resource, command, payload={}, silentErrors=False, exitOnError=False, exitMsg=None, returnType=ReturnType.json, endpoint=None):
+        if not endpoint:
+            endpoint = self.endpointHttp
+        cmd = "%s/v1/%s/%s" % (endpoint, resource, command)
```

Comment:
Should this be an f string?

Reply:
I think so.  They really are so much more readable.  It did make it easy to inspect the diff, but that's only momentarily useful.